tests/integration_tests/functional/test_max_vcpus.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests scenario for microvms with max vcpus(32)."""
+import os
+import select
+from framework.state_machine import TestState
+
+MAX_VCPUS = 32
+
+
+class WaitCPUs(TestState):
+    """Initial state when we wait for the login prompt."""
+
+    def handle_input(self, microvm, input_char) -> TestState:
+        """Handle input and return next state."""
+        if self.match(input_char):
+            return TestFinished()
+        return self
+
+
+class TestFinished(TestState):
+    """Test complete and successful."""
+
+    def handle_input(self, microvm, input_char) -> TestState:
+        """Return self since the test is about to end."""
+        return self
+
+
+def test_max_vcpus(test_microvm_with_ssh):
+    """Test if all configured vcpus are brought up at boot."""
+    microvm = test_microvm_with_ssh
+    microvm.jailer.daemonize = False
+    microvm.spawn()
+
+    # Configure a microVM with 32 vCPUs and a serial console.
+    microvm.basic_config(vcpu_count=MAX_VCPUS,
+                         boot_args='console=ttyS0 reboot=k panic=1 pci=off')
+
+    microvm.start()
+
+    # Setup screen log
+    screen_log = "/tmp/screen.log"
+    screen_log_fd = os.open(screen_log, os.O_RDONLY)
+
+    poller = select.poll()
+
+    # Set initial state - wait for cpus boot string
```

Comment:
This is probably unrelated to your PR, but should we actually check the online cpus == configured cpus and that offline cpus == 0? I am just wondering if we are leaking host topology information here (i.e. if the total number of CPUs of the host - available CPUs shows up as offline CPUs).

Reply:
We are not leaking CPU information here, the only CPUs the guest is seeing is the vcpus presented by fdt and there should be no relation to the actual host cpu. As an experiment I started a vm with 128 cpus on a host with 32 physical cpus and had no issues.